code/game/objects/structures/spirit_board.dm

```diff
@@ -32,24 +32,22 @@
 		virgin = 0
 		notify_ghosts("Someone has begun playing with a [src.name] in [get_area(src)]!", source = src)
 
-	planchette = input("Choose the letter.", "Seance!") in list("A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z")
+	planchette = input("Choose the letter.", "Seance!") in null|anything in list("A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z")
+	if(!planchette || !Adjacent(M) || next_use > world.time)
+		return
 	add_logs(M, src, "picked a letter on", " which was \"[planchette]\".")
-	cooldown = world.time
+	next_use = world.time + rand(30,50)
 	lastuser = M.ckey
-
-	var/turf/T = loc
-	sleep(rand(20,30))
-	if(T == loc)
-		visible_message("<span class='notice'>The planchette slowly moves... and stops at the letter \"[planchette]\".</span>")
-
+	//blind message is the same because not everyone brings night vision to seances
+	visible_message("<span class='notice'>The planchette slowly moves... and stops at the letter \"[planchette]\".</span>","","<span class='notice'>The planchette slowly moves... and stops at the letter \"[planchette]\".</span>")
```

Comment:
default behaviour is for all special messages (self, blind) to be the basic message, this duplication is needless.